R/url_path.R

```diff
@@ -0,0 +1,20 @@
+#' Construct URL Path
+#' 
+#' Construct a URL from path components like \code{file.path}
+#'
+#' @param x a URL, or vector of URLs
+#' @param ... character vectors
+#' 
+#' @examples
+#' 
+#' url_path("https://github.com", c('Ironholds', 'nfultz'), 'urltools')
+#' 
+#' @export
+#'
+url_path <- function(x, ...) {
+  p <- path(x)
+  p[is.na(p)] = '.'
```

Comment:
It's a workaround to keep from propagating the NAs. `/./` then gets deleted by the gsub. I think this is pretty ugly and would like a better solution, though.